pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepositoryUtils.java

```diff
@@ -344,41 +345,36 @@ public static void recursivelyDeleteEmptyFolder(File element) {
     }
 
     /**
-     * Method to transform a Maven identifier into a filesystem path
+     * Method to transform an artifact file path to Maven artifact identifier format
      *
-     * Transform <group-id>:<artifact-id>:<packaging>:<version> to
-     * <group-id>/<artifact-id>/<version>/<artifact-id>-<version>[-classifier].<packaging>
+     * Transforms <...>/<group-id>/<artifact-id>/<version>/<artifact-id>-<version>[-classifier].<packaging> to
+     * <group-id>:<artifact-id>:<packaging>:<version>[:classifier]
      *
-     * @param identifier
-     * @return
+     * @param artifactPath - artifact file path
+     * @return Maven artifact identifier
      */
-    public static String convertMavenIdentifierToPathRegex(String identifier) {
-        String[] sections = identifier.split(":");
-        if (sections.length != 4 && sections.length != 5) {
-            throw new IllegalArgumentException(
-                    identifier
-                            + " is wrongly formatted! It should be <group-id>:<artifact-id>:<packaging>:<version>:[classifier]");
-        }
-        String fileSeparator = System.getProperty("file.separator");
+    public static String convertArtifactPathToIdentifier(String artifactPath) {
+        var fileSeparator = System.getProperty("file.separator");
```

Comment:
Please change this to `File.separator` , we tend to use that when accessing the default separator on other places, also by doing so you can get rid of this variable and use it directly on next line.

Reply:
I had to do it like this: `Pattern.quote(File.separator)`
Because on Windows, file separator is a `\` which is not a valid regex